parse/lex.go

```diff
@@ -157,16 +164,92 @@ func (s *state) acceptRun(valid string) {
 
 }
 
-func DefaultStateFn(s *state, e emitter) stateFn {
+//
+//	Bang     // !
+//  Assign   // =
+//
+//	Plus     // +
+//	Minus    // -
+//	Asterisk // *
+//	Slash    // /
+//	Mod      // %
+//
+//	LT     // <
+//	GT     // >
+//	LTE    // <=
+//	GTE    // >=
+//	EQ     // ==
+//	NOT_EQ // !=
+//
+//	Comma // ,
+//
+//	Lparen // (
+//	Rparen // )
+//	Lbrace // {
+//	Rbrace // }
+//
+// TODO: LTE, GTE, spaceStateFn, numberStateFn, identifierStateFn case
+func defaultStateFn(s *state, e emitter) stateFn {
+
+	switch ch := s.next(); {
+
+	case ch == '!':
+		if s.peek() == '=' {
+			s.next()
```

Comment:
How about refactoring this logic to another func?

Reply:
Because this is so simple logic,
i think this seems to be more understandable!